Discovery Service with AWS instance

nagulmeeras opened the pull request.

naveed050 committed:
Initial commit for CatalogService. Can add/get images

naveed050 committed:
Initial commit for CatalogService. Can add/get images

naveed050 committed:
Initial commit for CatalogService. Can add/get images

naveed050 committed:
Initial commit for CatalogService. Can add/get images

naveed050 committed:
Initial commit for CatalogService. Can add/get images

naveed050 committed:
code changes for catalog service

naveed050 committed:
code changes for catalog service

naveed050 committed:
code changes for catalog service

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added changes to delete a node

naveed050 committed:
added changes to delete a node

naveed050 committed:
added changes to delete a node

naveed050 committed:
added changes to delete a node

naveed050 committed:
added changes to delete a node

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added few modifications to Catalog Service

naveed050 committed:
added few modifications to Catalog Service

Commit:
written few endpoints(save,delete,get) for appsettings

Commit:
written few end points(save,delete,get) for appsettings

Commit:
written few end points(save,delete,get) for appsettings

Commit:
written few end points(save,delete,get) for appsettings

Commit:
written few end points(save,delete,get) for appsettings

Commit:
written few end points(save,delete,get) for appsettings

Commit:
made some changes and written app setting end points

Commit:
made some changes and written app setting end points

Commit:
made some changes and written app setting end points

Commit:
made some changes and written app setting end points

Commit:
made some changes and written app setting end points

Commit:
made some changes and written app setting end points

naveed050 committed:
added implicit classes

naveed050 committed:
added implicit classes

naveed050 committed:
added implicit classes

naveed050 committed:
added implicit classes

naveed050 committed:
added implicit classes

naveed050 committed:
added fromGraph and toGraph for a test case class

naveed050 committed:
added fromGraph and toGraph for a test case class

naveed050 committed:
added fromGraph and toGraph for a test case class

naveed050 committed:
added fromGraph and toGraph for a test case class

naveed050 committed:
added fromGraph and toGraph for a test case class

Commit:
Wriiten toGraph and fromGraph implementaions for saving and retrieving

Commit:
Wriiten toGraph and fromGraph implementaions for saving and retrieving

Commit:
Wriiten toGraph and fromGraph implementaions for saving and retrieving

Commit:
Wriiten toGraph and fromGraph implementaions for saving and retrieving

Commit:
Changes made according to portal implementaion in neo4j operations

Commit:
Changes made according to portal implementaion in neo4j operations

Commit:
Changes made according to portal implementaion in neo4j operations

Commit:
Changes made according to portal implementaion in neo4j operations

naveed050 committed:
added toGraph and fromGraph for Site and its related caseclasses

naveed050 committed:
added toGraph and fromGraph for Site and its related caseclasses

naveed050 committed:
added toGraph and fromGraph for Site and its related caseclasses

naveed050 committed:
added toGraph and fromGraph for Site and its related caseclasses

naveed050 committed:
added toGraph and fromGraph for Site and its related caseclasses

nagulmeeras committed:
Changes made according to the design in neo4j repositories

naveed050 committed:
Catalog Services

naveed050 committed:
Catalog Services

naveed050 committed:
Catalog Services

naveed050 committed:
Catalog Services

naveed050 committed:
Catalog Services

nagulmeeras committed:
Changed service routes from onComplete to onSuccess

naveed050 committed:
Added code to catch exceptions

naveed050 committed:
Added code to catch exceptions

naveed050 committed:
Added code to catch exceptions

naveed050 committed:
Added code to catch exceptions

nagulmeeras committed:
Changes made for handling future failure and success response

nagulmeeras committed:
Changes made for handling future failure and success response

nagulmeeras committed:
Changes made for handling future failure and success response

nagulmeeras committed:
Changes made for handling future failure and success response

naveed050 committed:
Instance case class

naveed050 committed:
Instance case class

naveed050 committed:
Instance case class

naveed050 committed:
Instance case class

naveed050 committed:
Instance case class

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

naveed050 committed:
Catalog Service after making changes

nagulmeeras committed:
Changes made according to review comments

naveed050 committed:
made changes to create graph with optional entities for some fields

naveed050 committed:
made changes to create graph with optional entities for some fields

naveed050 committed:
made changes to create graph with optional entities for some fields

naveed050 committed:
made changes to create graph with optional entities for some fields

naveed050 committed:
made changes to create graph with optional entities for some fields

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
added NodeService

naveed050 committed:
changed return type from list to Page

naveed050 committed:
changed return type from list to Page

naveed050 committed:
changed return type from list to Page

naveed050 committed:
changed return type from list to Page

naveed050 committed:
added modifications to NodeService

naveed050 committed:
added modifications to NodeService

naveed050 committed:
added modifications to NodeService

naveed050 committed:
added modifications to NodeService

naveed050 committed:
added modifications to NodeService

nagulmeeras committed:
Changed repository files for removing dummy objects

nagulmeeras committed:
APMService implementation initial commit

nagulmeeras committed:
APMService implementation initial commit

naveed050 committed:
Made few format changes

naveed050 committed:
Made few format changes

naveed050 committed:
Made few format changes

naveed050 committed:
Made few format changes

nagulmeeras committed:
Changes made according to comments

nagulmeeras committed:
Changes made according to comments

nagulmeeras committed:
Changes made according to comments

nagulmeeras committed:
Changes made according to comments

nagulmeeras committed:
Changes made according to comments

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

naveed050 committed:
refactored code by removing service layer

nagulmeeras committed:
Removed service layer and modified throwing exception by None

nagulmeeras committed:
Removed service layer and modified throwing exception by None

nagulmeeras committed:
Removed service layer and modified throwing exception by None

nagulmeeras committed:
Removed service layer and modified throwing exception by None

nagulmeeras committed:
Removed service layer and modified throwing exception by None

naveed050 committed:
changed as per PR review

naveed050 committed:
changed as per PR review

naveed050 committed:
changed as per PR review

naveed050 committed:
changed as per PR review

naveed050 committed:
changes related to error logging

naveed050 committed:
changes related to error logging

naveed050 committed:
changes related to error logging

nagulmeeras committed:
Changes made according to review comments

nagulmeeras committed:
Changes made according to review comments

nagulmeeras committed:
Changes made according to review comments

nagulmeeras committed:
Changes made according to review comments

nagulmeeras committed:
Changes made according to review comments

naveed050 committed:
fixed format and imported packages to avoid unnecessary typing everytime

naveed050 committed:
fixed format and imported packages to avoid unnecessary typing everytime

naveed050 committed:
fixed format and imported packages to avoid unnecessary typing everytime

naveed050 committed:
fixed format and imported packages to avoid unnecessary typing everytime

nagulmeeras committed:
Changes made according comments in controller and model classes

nagulmeeras committed:
Changes made according comments in controller and model classes

nagulmeeras committed:
Changes made according comments in controller and model classes

nagulmeeras committed:
Changes made according comments in controller and model classes

nagulmeeras committed:
Changes made according comments in controller and model classes

naveed050 committed:
using foldLeft instead of map and filter

naveed050 committed:
using foldLeft instead of map and filter

naveed050 committed:
using foldLeft instead of map and filter

naveed050 committed:
using foldLeft instead of map and filter

nagulmeeras committed:
Changed code in models to remove mutable collections

nagulmeeras committed:
Changed code in models to remove mutable collections

nagulmeeras committed:
Changed code in models to remove mutable collections

nagulmeeras committed:
Changed code in models to remove mutable collections

nagulmeeras committed:
Changed code in models to remove mutable collections

nagulmeeras committed:
Made changes in models to remove unused statements

nagulmeeras committed:
Made changes in models to remove unused statements

nagulmeeras committed:
Made changes in models to remove unused statements

nagulmeeras committed:
Made changes in models to remove unused statements

nagulmeeras committed:
Made changes in models to remove unused statements

nagulmeeras committed:
Changed folder structure

nagulmeeras committed:
Changed folder structure

nagulmeeras committed:
Changed folder structure

naveed050 committed:
refactored code

naveed050 committed:
refactored code

naveed050 committed:
refactored code

naveed050 committed:
refactored code

naveed050 committed:
refactored code

naveed050 committed:
changed directory structure

naveed050 committed:
changed directory structure

nagulmeeras committed:
Modified for merging

nagulmeeras committed:
Modified for merging

nagulmeeras committed:
Modified for merging

nagulmeeras committed:
written few endpoints(save,delete,get) for appsettings

nagulmeeras committed:
written few endpoints(save,delete,get) for appsettings

nagulmeeras committed:
written few endpoints(save,delete,get) for appsettings

nagulmeeras committed:
Changes made according to the design in neo4j repositories

nagulmeeras committed:
Changes made according to the design in neo4j repositories

nagulmeeras committed:
Changes made according to the design in neo4j repositories

nagulmeeras committed:
Changed service routes from onComplete to onSuccess

nagulmeeras committed:
Changed service routes from onComplete to onSuccess

nagulmeeras committed:
Changed service routes from onComplete to onSuccess

nagulmeeras committed:
Changes made according to review comments

nagulmeeras committed:
Changes made according to review comments

nagulmeeras committed:
Changes made according to review comments

nagulmeeras committed:
Changed repository files for removing dummy objects

nagulmeeras committed:
Changed repository files for removing dummy objects

nagulmeeras committed:
Changed repository files for removing dummy objects

nagulmeeras committed:
APMService implementation initial commit

nagulmeeras committed:
APMService implementation initial commit

nagulmeeras committed:
APMService implementation initial commit

nagulmeeras committed:
Changed folder structure

nagulmeeras committed:
Reformated code after merging

nagulmeeras committed:
Reformated code after merging

nagulmeeras committed:
Removed log4j depenedency and end points related to it

nagulmeeras committed:
Removed log4j depenedency and end points related to it

nagulmeeras committed:
Removed log4j depenedency and end points related to it

naveed050 committed:
modified gitignore

naveed050 committed:
rebased and resolved conflicts

naveed050 committed:
rebased and resolved conflicts

nagulmeeras committed:
Resolved conflicts after rebasing

nagulmeeras committed:
Resolved conflicts after rebasing

nagulmeeras committed:
Merge branch 'master' of https://github.com/nagulmeeras/activeGrid

Conflicts:
	src/main/scala/com/imaginea/Main.scala
	src/main/scala/com/imaginea/activegrid/core/models/Site.scala

naveed050 committed:
Added Discovery Service

nagulmeeras committed:
Merge branch 'SiteService' of https://github.com/nagulmeeras/activeGrid

Conflicts:
	src/main/scala/com/imaginea/Main.scala
	src/main/scala/com/imaginea/activegrid/core/models/GraphDBExecutor.scala
	src/main/scala/com/imaginea/activegrid/core/models/InstanceConnection.scala
	src/main/scala/com/imaginea/activegrid/core/models/Site.scala

naveed050 committed:
Added DiscoveryService

naveed050 committed:
Added DiscoveryService

naveed050 committed:
Discovery Service , Added code to create Instance from awsInstance and wrote custom read and write for jsonFormat as it has more than 22 fields

naveed050 committed:
removing graph db folder

nagulmeeras committed:
Removed mutable list and var in models

nagulmeeras committed:
Added Reserved instance details in site model

naveed050 committed:
Added LoadBalancer and ScalingGroup to Site

nagulmeeras committed:
Changed reserved instance details model

naveed050 committed:
Modified custom write of Instance case class

nagulmeeras committed:
Written toGraph and fromGraph implementaions and end point to get regions

naveed050 committed:
added toGraph and fromGraph of LoadBalancer and ScalingGroup

nagulmeeras committed:
Written GroupInstance models and end point

nagulmeeras committed:
Written getInstanceGroup and saveInstanceGroup endpoints

naveed050 committed:
Added 2 end points getKeyNames and getSiteView

eklavya commented on src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala:
```diff
@@ -69,24 +69,22 @@ object APMServerDetails {
+              (tuple, relationsip) =>
+                val childNode = relationsip.getEndNode
+                relationsip.getType.name match {
+                  case `apmServer_site_relation` => Tuple2(Site.fromNeo4jGraph(childNode.getId), tuple._2)
```
You don't actually have to call Tuple2 you ca just write (a,b). Same goes for type signature as well.

nagulmeeras replied:
Usage of explicit Tuple() signature is removed

eklavya replied:
removed where?

nagulmeeras replied:
Removed it and pushed the code now

eklavya commented on src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala:
```diff
@@ -0,0 +1,192 @@
+      val awsInstancesResult = getAWSInstances(amazonEC2)
+      val totalsecurityGroups = amazonEC2.describeSecurityGroups.getSecurityGroups.foldLeft(Map[String, SecurityGroup]())((map, sg) => map + ((sg.getGroupId, sg)))
+      val addresses = amazonEC2.describeAddresses.getAddresses.foldLeft(Map[String, Address]())((map, address) => map + ((address.getInstanceId, address)))
+      val imageIds = awsInstancesResult.foldLeft(List[String]())((list, awsInstance) => list.::(awsInstance.getImageId))
```
Why list.:: ?

nagulmeeras replied:
for adding imageId of each AWSInstance to List.
Should i change it like this  list++List(valueToBeAdded) ?

eklavya replied:
I mean :: without the preceding dot.

nagulmeeras replied:
Removed .:: and used :: in place now

nagulmeeras committed:
Merge branch 'DiscoveryService' of https://github.com/naveed050/activeGrid

Conflicts:
	.gitignore
	src/main/scala/com/imaginea/Main.scala
	src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala
	src/main/scala/com/imaginea/activegrid/core/models/AWSContextBuilder.scala
	src/main/scala/com/imaginea/activegrid/core/models/AWSInstanceType.scala
	src/main/scala/com/imaginea/activegrid/core/models/AccountInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/Filter.scala
	src/main/scala/com/imaginea/activegrid/core/models/FilterType.scala
	src/main/scala/com/imaginea/activegrid/core/models/GraphDBExecutor.scala
	src/main/scala/com/imaginea/activegrid/core/models/ImageInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/Instance.scala
	src/main/scala/com/imaginea/activegrid/core/models/InstanceBlockDeviceMappingInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/InstanceProvider.scala
	src/main/scala/com/imaginea/activegrid/core/models/IpPermissionInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/SecurityGroupInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/Site.scala
	src/main/scala/com/imaginea/activegrid/core/models/Site1.scala
	src/main/scala/com/imaginea/activegrid/core/models/SiteFilter.scala
	src/main/scala/com/imaginea/activegrid/core/models/SnapshotInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/ViewType.scala
	src/main/scala/com/imaginea/activegrid/core/models/VolumeInfo.scala

naveed050 committed:
Added filterInstance methods

nagulmeeras committed:
Removed explicit Tuple() signature in models

eklavya commented on src/main/scala/com/imaginea/activegrid/core/models/SnapshotInfo.scala:
```diff
@@ -0,0 +1,81 @@
+          }
+          Some(SnapshotInfo(
+            Some(nodeId),
+            repository.getProperty[String](node, "snapshotId").get,
```
all these .get is it impossible to get a None here?

nagulmeeras replied:
Handled the case if the node is not present

naveed050 committed:
Discovery service (#17)

* Initial commit for CatalogService. Can add/get images

* added few modifications to Catalog Service

* added changes to delete a node

* added few modifications to Catalog Service

* added implicit classes

* added fromGraph and toGraph for a test case class

* added toGraph and fromGraph for Site and its related caseclasses

* Catalog Services

* Added code to catch exceptions

* Instance case class

* Catalog Service after making changes

* Catalog Service after making changes

* made changes to create graph with optional entities for some fields

* added NodeService

* added NodeService

* changed return type from list to Page

* added modifications to NodeService

* Made few format changes

* refactored code by removing service layer

* refactored code by removing service layer

* refactored code by removing service layer

* changed as per PR review

* changes related to error logging

* fixed format and imported packages to avoid unnecessary typing everytime

* using foldLeft instead of map and filter

* refactored code

* changed directory structure

* Initial commit for CatalogService. Can add/get images

* code changes for catalog service

* added few modifications to Catalog Service

* added changes to delete a node

* added few modifications to Catalog Service

* added implicit classes

* added fromGraph and toGraph for a test case class

* added toGraph and fromGraph for Site and its related caseclasses

* Catalog Services

* Added code to catch exceptions

* Instance case class

* Catalog Service after making changes

* Catalog Service after making changes

* made changes to create graph with optional entities for some fields

* added NodeService

* added NodeService

* changed return type from list to Page

* added modifications to NodeService

* Made few format changes

* refactored code by removing service layer

* refactored code by removing service layer

* refactored code by removing service layer

* changed as per PR review

* changes related to error logging

* fixed format and imported packages to avoid unnecessary typing everytime

* using foldLeft instead of map and filter

* refactored code

* rebased and resolved conflicts

* Added Discovery Service

* Added DiscoveryService

* Discovery Service , Added code to create Instance from awsInstance and wrote custom read and write for jsonFormat as it has more than 22 fields

* removing graph db folder

* Added LoadBalancer and ScalingGroup to Site

* Modified custom write of Instance case class

* added toGraph and fromGraph of LoadBalancer and ScalingGroup

* Added 2 end points getKeyNames and getSiteView

* Added filterInstance methods

nagulmeeras committed:
Merge branch 'DiscoveryService' of https://github.com/naveed050/activeGrid
And Removed adding list with dot
Conflicts:
	src/main/scala/com/imaginea/Main.scala
	src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala

naveed050 committed:
Discovery service (#21)

* Initial commit for CatalogService. Can add/get images

* added few modifications to Catalog Service

* added changes to delete a node

* added few modifications to Catalog Service

* added implicit classes

* added fromGraph and toGraph for a test case class

* added toGraph and fromGraph for Site and its related caseclasses

* Catalog Services

* Added code to catch exceptions

* Instance case class

* Catalog Service after making changes

* Catalog Service after making changes

* made changes to create graph with optional entities for some fields

* added NodeService

* added NodeService

* changed return type from list to Page

* added modifications to NodeService

* Made few format changes

* refactored code by removing service layer

* refactored code by removing service layer

* refactored code by removing service layer

* changed as per PR review

* changes related to error logging

* fixed format and imported packages to avoid unnecessary typing everytime

* using foldLeft instead of map and filter

* refactored code

* changed directory structure

* Initial commit for CatalogService. Can add/get images

* code changes for catalog service

* added few modifications to Catalog Service

* added changes to delete a node

* added few modifications to Catalog Service

* added implicit classes

* added fromGraph and toGraph for a test case class

* added toGraph and fromGraph for Site and its related caseclasses

* Catalog Services

* Added code to catch exceptions

* Instance case class

* Catalog Service after making changes

* Catalog Service after making changes

* made changes to create graph with optional entities for some fields

* added NodeService

* added NodeService

* changed return type from list to Page

* added modifications to NodeService

* Made few format changes

* refactored code by removing service layer

* refactored code by removing service layer

* refactored code by removing service layer

* changed as per PR review

* changes related to error logging

* fixed format and imported packages to avoid unnecessary typing everytime

* using foldLeft instead of map and filter

* refactored code

* rebased and resolved conflicts

* Added Discovery Service

* Added DiscoveryService

* Discovery Service , Added code to create Instance from awsInstance and wrote custom read and write for jsonFormat as it has more than 22 fields

* removing graph db folder

* Added LoadBalancer and ScalingGroup to Site

* Modified custom write of Instance case class

* added toGraph and fromGraph of LoadBalancer and ScalingGroup

* Added 2 end points getKeyNames and getSiteView

* Added filterInstance methods

* modified toGraph and fromGraph to stop saving primitives as NOVAL when there is no value, removed GraphDBExecutor and placed all db related functions in a single file Neo4jRepository

nagulmeeras committed:
Changed tuple naming convention in models and main

nagulmeeras committed:
Conflicts resolved after rebasing

Conflicts:
	.gitignore
	src/main/scala/com/imaginea/Main.scala
	src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala
	src/main/scala/com/imaginea/activegrid/core/models/AWSContextBuilder.scala
	src/main/scala/com/imaginea/activegrid/core/models/AWSInstanceType.scala
	src/main/scala/com/imaginea/activegrid/core/models/AccountInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/Filter.scala
	src/main/scala/com/imaginea/activegrid/core/models/FilterType.scala
	src/main/scala/com/imaginea/activegrid/core/models/GraphDBExecutor.scala
	src/main/scala/com/imaginea/activegrid/core/models/ImageInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/InstanceBlockDeviceMappingInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/InstanceProvider.scala
	src/main/scala/com/imaginea/activegrid/core/models/IpPermissionInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/Neo4jRepository.scala
	src/main/scala/com/imaginea/activegrid/core/models/ScalingGroup.scala
	src/main/scala/com/imaginea/activegrid/core/models/SecurityGroupInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/Site.scala
	src/main/scala/com/imaginea/activegrid/core/models/Site1.scala
	src/main/scala/com/imaginea/activegrid/core/models/SiteFilter.scala
	src/main/scala/com/imaginea/activegrid/core/models/SnapshotInfo.scala
	src/main/scala/com/imaginea/activegrid/core/models/Software.scala
	src/main/scala/com/imaginea/activegrid/core/models/ViewType.scala
	src/main/scala/com/imaginea/activegrid/core/models/VolumeInfo.scala

nagulmeeras committed:
Resolved confilct In repository file

nagulmeeras committed:
Resolved confilcts in Neo4jRepository File

nagulmeeras committed:
Removed .:: usage to add list in Controller

nagulmeeras committed:
Removed Tuple signature and AnyRef usage in models

nagulmeeras committed:
Used type safe lib for loading dbpath from application config

nagulmeeras committed:
Changed port value
